cpp/example_code/dynamodb/get_item.cpp

```diff
@@ -1,92 +1,118 @@
+  \param clientConfiguration: AWS client configuration.
+  \return bool: Function succeeded.
+ */
 
-/*
-   Get an item from a DynamoDB table.
+bool AwsDoc::DynamoDB::getItem(const Aws::String &tableName,
+                               const Aws::String &partitionKey,
+                               const Aws::String &partitionValue,
+                               const Aws::Client::ClientConfiguration &clientConfiguration) {
+    Aws::DynamoDB::DynamoDBClient dynamoClient(clientConfiguration);
+    Aws::DynamoDB::Model::GetItemRequest request;
 
-   To run this C++ code example, ensure that you have setup your development environment, including your credentials.
-   For information, see this documentation topic:
-   https://docs.aws.amazon.com/sdk-for-cpp/v1/developer-guide/getting-started.html
-*/
+    // Set up the request.
+    request.SetTableName(tableName);
+    request.AddKey(partitionKey,
+                   Aws::DynamoDB::Model::AttributeValue().SetS(partitionValue));
 
-int main(int argc, char** argv)
-{
-    const std::string USAGE = "\n" \
-        "Usage: <tableName> <key> <keyval> \n" 
-        "Where:\n" 
-        "    tableName - the Amazon DynamoDB table from which an item is retrieved (for example, Music3). \n"
-        "    key - the key used in the Amazon DynamoDB table (for example, Artist). \n" 
-        "    keyval - the key value that represents the item to get (for example, Famous Band).\n";
+    // Retrieve the item's fields and values.
+    const Aws::DynamoDB::Model::GetItemOutcome &outcome = dynamoClient.GetItem(request);
+    if (outcome.IsSuccess()) {
+        // Reference the retrieved fields/values.
+        const Aws::Map<Aws::String, Aws::DynamoDB::Model::AttributeValue> &item = outcome.GetResult().GetItem();
+        if (!item.empty()) {
+            // Output each retrieved field and its value.
+            for (const auto &i: item)
+                std::cout << "Values: " << i.first << ": " << i.second.GetS()
+                          << std::endl;
+        }
+        else {
+            std::cout << "No item found with the key " << partitionKey << std::endl;
+        }
+    }
+    else {
+        std::cerr << "Failed to get item: " << outcome.GetError().GetMessage();
+    }
 
-    if (argc < 3)
-    {
-        std::cout << USAGE;
+    return outcome.IsSuccess();
+}
+// snippet-end:[dynamodb.cpp.get_item.code]
+
+/*
+ *
+ *  main function
+ *
+ *  Usage: 'run_get_item <table_name> <partition_key> <partition_value>'
+ *
+ *  Prerequisites: a DynamoDB table named <table_name> containing an item with
+ *     <partition_value> for its <partition_key>.
+ *
+ */
+
+#ifndef TESTING_BUILD
+
+int main(int argc, char **argv) {
+    if (argc < 4) {
+        std::cout << R"("Usage:
+    run_get_item <table_name> <partition_key> <partition_value>
+Where:
+    table - the table to get the item from.
+    partition_key  - the partition key of the table,
+    partition_value - the item value for the partition key"

```

Comment:
table - The table to get the item from.
    partition_key  - The partition key of the table.
    partition_value - The item value for the partition key.